ReplayMessenger.device.lib.nut

```diff
@@ -93,6 +109,41 @@ class ReplayMessenger {
         return _mm.send(messageName, data, null, _retryInterval, metadata);
     }
 
+    function onReplayBegin(loggerName, callback = null) {
+      if (callback == null && typeof loggerName == "string") { //de-register callback
+        if (loggerName in _onReplayBegin) {
+          delete _onReplayBegin[loggerName]
+        }
+      } else if (callback == null) { //assume callback is provided for "default" logger
+        callback = loggerName;
+        loggerName = "default";
+      } else {
+        _onReplayBegin[loggerName] <- callback;
+      }
+    }
+
+    function onReplayComplete(loggerName, callback = null) {
+      if (callback == null && typeof loggerName == "string") { //de-register callback
+        if (loggerName in _onReplayComplete) {
+          delete _onReplayComplete[loggerName]
+        }
+      } else if (callback == null) { //assume callback is provided for "default" logger
+        callback = loggerName;
+        loggerName = "default";
+      } else {
+        _onReplayComplete[loggerName] <- callback;
+      }
+    }
+
+    function setLoggerReadOptions(loggerName, options=null) {
+      if (typeof loggerName == "table" && options == null) {
```

Comment:
This is a vicious practice to dynamically switch positions of the arguments! Let's please not do so!

a) it is very confusion and 
b) `setLoggerReadOptions(null, {...});` would make it fail.

If you want to have named optional parameters let's just use a map with two values. But I don't really think it's worth it.